### DISCUSSION TITLE
Unable to use Terminal

### DISCUSSION THREAD

**Discussion by IamSmeagol** [folke/snacks.nvim#67] [Q&A]
When I launch the terminal using `<leader>ft` I only see a blank window. Manually launching with `:lua Snacks.terminal.toggle()` results in the same. Any advice is appreciated. 

See errors [here](https://gist.github.com/IamSmeagol/8c349232d0f8c4485e097100a574c495)

![image](https://github.com/user-attachments/assets/734133fe-1467-4589-8762-980f6b3303ac)

Using Lazyvim 13.2.0, Snacks as of commit f704f74, neovim v0.11.0-dev-1137+g72a1df6065 , on Windows 11.  

https://github.com/folke/snacks.nvim/discussions/58 this looked similar, but the problem isn't solved for me outside of gitbash

**Comment by iamxiaojianzheng** [comment]
Try the configuration?

```lua
vim.o.shell = "bash"
vim.o.shellcmdflag = "-i"
```

**Reply by IamSmeagol** [reply]
I tried adding it, but it did not work. I did already have `vim.opt.shell = "bash"` as part of my config prior to making this, but changing the shellcmdflag didn't work for me. Thank you for suggesting though :)

**Comment by folke** [comment]
There's another related issue I'm trying to solve.

Can you give me the exact output of `:=vim.o.shell`?

**Reply by folke** [reply]
I just pushed an update to fix that other issue. Can you verify everything still works for you as well?

**Reply by IamSmeagol** [reply]
I got `bash` as the output from that command --  Checking the shell command flag I get `vim.o.shellcmdflag /s /c`

I still get this error after updating to the latest neovim and snacks install
```
   Error  9:30:21 AM msg_show.emsg E5108: Error executing lua: ...Local/nvim-data/lazy/snacks.nvim/lua/snacks/terminal.lua:90: Error executing lua: Vim:E475: Invalid value for argument cmd: 'bash' is not executable
stack traceback:
	[C]: in function 'termopen'
	...Local/nvim-data/lazy/snacks.nvim/lua/snacks/terminal.lua:95: in function <...Local/nvim-data/lazy/snacks.nvim/lua/snacks/terminal.lua:90>
	[C]: in function 'nvim_buf_call'
	...Local/nvim-data/lazy/snacks.nvim/lua/snacks/terminal.lua:90: in function 'open'
	...Local/nvim-data/lazy/snacks.nvim/lua/snacks/terminal.lua:131: in function 'terminal'
	...al/nvim-data/lazy/LazyVim/lua/lazyvim/config/keymaps.lua:160: in function <...al/nvim-data/lazy/LazyVim/lua/lazyvim/config/keymaps.lua:160>
stack traceback:
	[C]: in function 'nvim_buf_call'
	...Local/nvim-data/lazy/snacks.nvim/lua/snacks/terminal.lua:90: in function 'open'
	...Local/nvim-data/lazy/snacks.nvim/lua/snacks/terminal.lua:131: in function 'terminal'
	...al/nvim-data/lazy/LazyVim/lua/lazyvim/config/keymaps.lua:160: in function <...al/nvim-data/lazy/LazyVim/lua/lazyvim/config/keymaps.lua:160>
   Error  9:30:21 AM msg_show.emsg E121: Undefined variable: b:term_title
   Error  9:30:21 AM msg_show.emsg E121: Undefined variable: b:term_title
```

**Reply by folke** [reply]
It seems that `bash` is not on your path

**Reply by IamSmeagol** [reply]
I am able to open `bash` when in normal powershell, but not while in neovim. 
I tried defining an absolute path `C:\\Users\\thesm\\AppData\\Local\\Microsoft\\WindowsApps\\bash.exe`
My environment variable includes `C:\Users\thesm\AppData\Local\Microsoft\WindowsApps` so bash should be included right?

I tried changing vim.opt.shell to powershell.exe and the terminal does work for both wsl and windows. And I can execute bash from there. I have no idea why I can't just start with bash though 😓 


Update I decided to us `jit.os` to chose between powershell if I'm on windows, and bash if on linux (wsl) it works nicely. I'll wait to close until bash works from windows though :)

```lua
if jit.os == "Windows" then
  vim.opt.shell = "powershell.exe"
else
  vim.opt.shell = "bash.exe"
end
```
